### PR TITLE
Added support for NoSQL databases

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: is_core/contrib/elasticsearch/views.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+class ElasticsearchTableView(ElasticsearchTableViewMixin, BaseModelTableView):
+
+    add_button_verbose_name = 'add %(verbose_name)s'
+
+
+class ElasticsearchDetailView(ModelReadonlyDetailView):
+
+    def _get_obj_or_none(self,):
```
Comment (rubickcz) [inline is_core/contrib/elasticsearch/views.py:48]: Trailing comma not needed

File: is_core/templates/is_core/forms/inline_table.html [hunk 2]
```diff
@@ -58,27 +58,31 @@
...
-                <li>
-                    <div class="dropdown columns-dropdown">
-                        <span class="btn btn-primary dropdown-toggle" data-toggle="dropdown">{% trans "Columns" %}<span class="caret"></span></span>
-                        <div class="dropdown-menu columns-manager" id="columns-manager-{{ table_slug }}"></div>
-                    </div>
-                </li>
-                {% endif %}
-            {% endblock %}
+			<ul class="inline-list">
+					{% block column-manager %}
```
Comment (rubickcz) [inline is_core/templates/is_core/forms/inline_table.html:62]: Too much indent?

File: is_core/generic_views/form_views.py [hunk 3]
```diff
@@ -223,9 +226,12 @@ def get_buttons_dict(self):
             }
         }
 
+    def get_fields(self):
+        return [] if self.fields is None else self.fields
```
Comment (dstlmrk) [inline is_core/generic_views/form_views.py:230]: What about this?
```python
return self.fields or []
```

File: is_core/generic_views/form_views.py [hunk 4]
```diff
@@ -576,37 +585,43 @@ def pre_save_obj(self, obj, form, change):
...
+        return self.core.get_fieldsets(self.request, self.get_obj(True))
 
     def get_fields(self):
-        return (self.fields is not None and self.fields or
-                self.core.get_ui_form_fields(self.request, self.get_obj(True)))
+        if self.fields is not None:
+            return self.fields
+
+        fields = self.core.get_fields(self.request, self.get_obj(True))
+        return () if fields is None else fields
```
Comment (dstlmrk) [inline is_core/generic_views/form_views.py:606]: What about this?
```python
return fields or ()
```

File: is_core/generic_views/inlines/base.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class RelatedInlineView(FieldPermissionViewMixin, InlineView):
+
+    permission = RelatedCoreAllowed()
+    model = None
+
+    def _get_field_permissions(self):
+        related_core = self._get_related_core()
+        if related_core:
```
Comment (dstlmrk) [inline is_core/generic_views/inlines/base.py:46]: I think this condition could be better written

```python
return self.field_permissions or (
    self.related_core.field_permissions if related_core else FieldsSetPermission()
)
```

File: example/dj/apps/issue_tracker/dynamodb/core.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+from is_core.contrib.dynamodb.cores import DynamoDBUIRESTISCore
+from is_core.patterns import RESTPattern, UIPattern, PK_PATTERN
+
+from .models import Comment
+
+
+class CommentPatternMixin:
+
+    def _get_try_kwargs(self, request, obj):
```
[the remark this answers is not in the document]
Comment (matllubos) [reply example/dj/apps/issue_tracker/dynamodb/core.py:11]: Uf this is complicated problem. I will explain it personally

File: example/dj/apps/issue_tracker/dynamodb/core.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    rest_range_key = 'user_id'
+
+    def _get_hash_key(self, request):
+        return request.kwargs.get('issue_pk')
+
+    def get_api_url(self, request):
+        return resolve_url(self.get_api_url_name(), issue_pk=self._get_hash_key(request))
+
+    def get_url_prefix(self):
```
[the remark this answers is not in the document]
Comment (matllubos) [reply example/dj/apps/issue_tracker/dynamodb/core.py:50]: No this is not as simple as may seen. You can use Global secondary index and hash key is not the same as hash key of the model. The second problem is that Dynamo model can have only hash key or hash key and range key. 

Maybe we can develop some complicated administration in the future. But this is out of scope right now. DynamoDB is not perfect right now, but slow steps in the library is better than nothing.

[the remark this answers is not in the document]
Comment (rubickcz) [reply example/dj/apps/issue_tracker/dynamodb/core.py:50]: OK

File: is_core/generic_views/form_views.py (hunk 4 above)
Comment (matllubos) [reply is_core/generic_views/form_views.py]: this is not the same 

fields is None != bool(fields)

Comment (dstlmrk) [reply is_core/generic_views/form_views.py]: I know but it depends what you expect in `fields`. Your solution is fine if you need detect exactly `None`.

**Pull request merged**